src/Middlewares/CacheResponse.php

```diff
@@ -27,18 +29,22 @@ public function handle(Request $request, Closure $next, ...$args): Response
         $tags = $this->getTags($args);
 
         if ($this->responseCache->enabled($request) && ! $this->responseCache->shouldBypass($request)) {
-            if ($this->responseCache->hasBeenCached($request, $tags)) {
-                event(new ResponseCacheHit($request));
+            try {
+                if ($this->responseCache->hasBeenCached($request, $tags)) {
+                    event(new ResponseCacheHit($request));
 
-                $response = $this->responseCache->getCachedResponseFor($request, $tags);
+                    $response = $this->responseCache->getCachedResponseFor($request, $tags);
 
-                $response = $this->addCacheAgeHeader($response);
+                    $response = $this->addCacheAgeHeader($response);
 
-                $this->getReplacers()->each(function (Replacer $replacer) use ($response) {
-                    $replacer->replaceInCachedResponse($response);
-                });
+                    $this->getReplacers()->each(function (Replacer $replacer) use ($response) {
+                        $replacer->replaceInCachedResponse($response);
+                    });
 
-                return $response;
+                    return $response;
+                }
+            } catch (CouldNotUnserialize $e) {
+                Log::error("Could not unserialize response, returning uncached response instead. Error: {$e->getMessage()}");
```

Comment:
Instead of logging this exception, use the `report()` function and also fire off a new event.

Could you also write a test for this and mention this behaviour in the readme?

Reply:
Will do. Haven't forgotten about this, just need some time to do it.

Reply:
@freekmurze I have replaced use of Log with report() and am firing a CacheMissed event, but I couldn't figure out the test..